nms/app/packages/magmalte/app/views/subscriber/SubscriberAddDialog.js

```diff
@@ -243,6 +246,75 @@ export function EditSubscriberButton(props: EditProps) {
   );
 }
 
+export function BulkDeleteSubscriberButton(props: {onClose: () => void}) {
+  const classes = useStyles();
+  const successCountRef = useRef(0);
+  const enqueueSnackbar = useEnqueueSnackbar();
+  const ctx = useContext(SubscriberContext);
+
+  const fileInput = useRef(null);
+  const deleteSubscribers = async subscribers => {
+    successCountRef.current = 0;
+    for (const subscriber of subscribers) {
+      try {
+        await ctx.setState?.(subscriber.imsi);
+        successCountRef.current = successCountRef.current + 1;
+      } catch (e) {
+        const errMsg = e.response?.data?.message ?? e.message ?? e;
+        enqueueSnackbar('error deleting ' + subscriber.imsi + ' : ' + errMsg, {
+          variant: 'error',
+        });
+        return;
+      }
+    }
+    enqueueSnackbar(
+      ` Subscriber${
+        successCountRef.current > 1 ? 's ' : ''
+      } deleted successfully`,
+      {
+        variant: 'success',
+      },
+    );
+    props.onClose();
+  };
+  return (
+    <>
+      <input
+        type="file"
+        ref={fileInput}
+        accept={'.csv'}
+        style={{display: 'none'}}
+        onChange={async () => {
+          if (fileInput.current?.files[0]) {
+            try {
+              const newSubscribers = await parseSubscriberFile({
+                fileObj: fileInput.current?.files[0],
+                delete: true,
+              });
+              deleteSubscribers(newSubscribers);
+            } catch (e) {
+              enqueueSnackbar(e, {
+                variant: 'error',
+              });
```

Comment:
We have a lot of problems with our snackbar error messages being rather cryptic to non-developers. If you can provide some easy to understand error messages for different failure modes here, that would improve the usability quite a bit.

Reply:
I think we should do this in another PR because it seems that we don't handle all the errors (for example deleting a non existing subscriber)